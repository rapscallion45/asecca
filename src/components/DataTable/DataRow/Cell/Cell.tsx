import { FC, memo } from 'react';
import { styled } from '@mui/material/styles';
import { TableCell } from '@mui/material';
import { tableCellClasses } from '@mui/material/TableCell';

/* table cell stylings */
const StyledTableCell = styled(TableCell)(({ theme }) => ({
  /* table head colors */
  [`&.${tableCellClasses.head}`]: {
    backgroundColor: theme.palette.common.black,
    color: theme.palette.common.white,
  },
  /* adjust font size */
  [`&.${tableCellClasses.body}`]: {
    fontSize: 14,
  },
}));

interface ICellProps {
  value: string;
  sx: any;
}

/* Data Table Cell helper component */
/* ================================ */
const Cell: FC<ICellProps> = (props) => {
  const { value, sx } = props;

  return (
    <StyledTableCell align="left" sx={sx}>
      {/* render passed value */}
      {value}
    </StyledTableCell>
  );
};

export default memo(Cell);
