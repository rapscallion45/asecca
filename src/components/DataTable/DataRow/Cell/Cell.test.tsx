import React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Table, TableBody, TableRow } from '@mui/material';
import Cell from './Cell';

/**
 * test text
 *
 * @since - 0.0.0
 */
const testValue = 'Device Processing';

/**
 * test style
 *
 * @since - 0.0.0
 */
const testSx = { fontWeight: 'bold' };

/**
 * Data Table Cell Unit Tests
 *
 * @author - Carl Scrivener {@link https://github.com/rapscallion45 GitHub}
 * @since - 0.0.0
 */
describe('Data Table Cell', () => {
  it('Renders correctly', async () => {
    /** perform snapshot test */
    const tree = renderer
      .create(
        <Table>
          <TableBody>
            <TableRow>
              <Cell value={testValue} sx={testSx} />
            </TableRow>
          </TableBody>
        </Table>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe('Value', () => {
    it('Should render passed value', async () => {
      /** Arrange */
      /** Act */
      render(
        <Table>
          <TableBody>
            <TableRow>
              <Cell value={testValue} sx={testSx} />
            </TableRow>
          </TableBody>
        </Table>
      );

      /** Assert */
      expect(screen.getByText(testValue)).toBeInTheDocument();
    });
  });

  describe('Styling', () => {
    it('Should render passed styling props', async () => {
      /** Arrange */
      /** Act */
      render(
        <Table>
          <TableBody>
            <TableRow>
              <Cell value={testValue} sx={testSx} />
            </TableRow>
          </TableBody>
        </Table>
      );

      /** Assert */
      expect(screen.getByText(testValue)).toBeInTheDocument();
    });
  });
});
