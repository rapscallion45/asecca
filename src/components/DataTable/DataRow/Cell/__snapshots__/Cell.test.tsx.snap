// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Data Table Cell Renders correctly 1`] = `
<table
  className="MuiTable-root css-rqglhn-MuiTable-root"
  role={null}
>
  <tbody
    className="MuiTableBody-root css-apqrd9-MuiTableBody-root"
    role={null}
  >
    <tr
      className="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
      role={null}
    >
      <td
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1nt70hm-MuiTableCell-root"
      >
        Device Processing
      </td>
    </tr>
  </tbody>
</table>
`;
